server/test/unit/common/config/test_app_config.py

```diff
@@ -0,0 +1,144 @@
+from server.common.config.app_config import AppConfig
+from server.test.unit.common.config import ConfigTests
+from server.common.errors import ConfigurationError
+from server.test import FIXTURES_ROOT
+
+
+class AppConfigTest(ConfigTests):
+    def setUp(self):
+        self.config_file_name = f"{unittest.TestCase.id(self).split('.')[-1]}.yml"
+        self.config = AppConfig()
+        self.config.update_server_config(multi_dataset__dataroot=FIXTURES_ROOT)
+        self.server_config = self.config.server_config
+        self.config.complete_config()
+
+        message_list = []
+
+        def noop(message):
+            message_list.append(message)
+
+        messagefn = noop
+        self.context = dict(messagefn=messagefn, messages=message_list)
+
+    def get_config(self, **kwargs):
+        file_name = self.custom_app_config(
+            dataroot=f"{FIXTURES_ROOT}", config_file_name=self.config_file_name, **kwargs)
+        config = AppConfig()
+        config.update_from_config_file(file_name)
+        return config
+
+    def test_get_default_config_successfully_reads_yaml_file(self):
+        app_default_config = AppConfig().get_default_config()
+
+        with open(f"{os.getenv('PROJECT_ROOT')}/default_config.yml", 'r') as default_config:
+            yaml_config = yaml.safe_load(default_config)
+
+        self.assertEqual(yaml_config['server']['app']['verbose'], app_default_config['server']['app']['verbose'])
+        self.assertEqual(yaml_config['server']['authentication']['type'],
+                         app_default_config['server']['authentication']['type'])
+
+    def test_init_app_config_pulls_in_default_server_and_dataset_configurations(self):
+        config = AppConfig()
+
+        with open(f"{os.getenv('PROJECT_ROOT')}/default_config.yml", 'r') as default_config:
+            yaml_config = yaml.safe_load(default_config)
+
+        self.assertEqual(yaml_config['server']['app']['verbose'], config.server_config.app__verbose)
+        self.assertEqual(yaml_config['server']['authentication']['type'], config.server_config.authentication__type)
+        self.assertEqual(yaml_config['dataset']['diffexp']['lfc_cutoff'],
+                         config.default_dataset_config.diffexp__lfc_cutoff)
+
+    def test_get_dataset_config_handles_single_datasets(self):
+        datapath = f"{FIXTURES_ROOT}/1e4dfec4-c0b2-46ad-a04e-ff3ffb3c0a8f.h5ad"
+        file_name = self.custom_app_config(
+            dataset_datapath=datapath, config_file_name=self.config_file_name)
+        config = AppConfig()
+        config.update_from_config_file(file_name)
+        self.assertEqual(config.server_config.single_dataset__datapath, datapath)
+
+    def test_update_server_config_updates_server_config_and_config_status(self):
+        config = self.get_config()
+        config.complete_config()
+        config.check_config()
+        config.update_server_config(multi_dataset__dataroot=FIXTURES_ROOT)
+        with self.assertRaises(ConfigurationError):
```

Comment:
The config error is because when you update the config the attribute has no longer been type checked and you need to call complete_config before check_config will be successful